### ISSUE TITLE
A linux update ?

### ISSUE THREAD

**Issue by Zbradaradjan** [Azurency/CQUI_Community-Edition#401]
Currently playing on Ubuntu 18.04, the mod doesn't work. Will it be updated to support the  v1.0.0.367 linux version ?

**Comment by Azurency** [comment]
Oh, so Linux is on a more advanced version number 👀. I didn't know that, the Mac OS and Windows version are both at 1.0.0.317, I know there's an update incoming in may, but... I don't think that's the one, was it updated recently ?

**Comment by Zbradaradjan** [comment]
Nope that's just a mistake when writing the version from me, apologies. It is 1.0.0.317. The mod doesn't work for me, so I figured it was maybe the linux version that wasn't updated.

**Comment by azorej** [comment]
Did mod ever work on linux correctly? I noticed you've lower cased files in the past to fix it on linux.
But there is a some strange behaviour. All standard game resources are lower cased but it doesn't seem to work for mod.
worldtracker_expansion1.lua <- opening/reading error
WorldTracker_Expansion1.lua <- works
minimappanel.lua <- works
MinimapPanel.lua <- opening/reading error
And worse, xml-replacements don't ever log opening/reading error.
As far as can see, game doesn't care about case in *.modinfo.
What i did for now: i made a script that makes symlinks for every mod resource (lower case, several variants of Camel notation). And it does work somehow.

Currently i'm trying to fix <s>buildings/units purchase in city panel</s> annoying great works window (it continues to spontaneously open)

**Comment by azorej** [comment]
Ok. Toggle great works window was bind to "w" by default. And i used WASD to control the screen. So there is no bug.

**Comment by Zbradaradjan** [comment]
Can you share your script please ? I'd like to see if it's the same issue on my end

**Comment by azorej** [comment]
https://github.com/azorej/CQUI_Community-Edition/tree/linux_fix

I made a script for convenience:
`curl https://raw.githubusercontent.com/azorej/CQUI_Community-Edition/linux_fix/checkout_mod.sh | bash -s`
You can pass custom path for mod files to it
`curl https://raw.githubusercontent.com/azorej/CQUI_Community-Edition/linux_fix/checkout_mod.sh | bash -s ~/games/other_mods`

You will need **python3** and **python3-pip** packages (you can install it via apt-get)
Please, unsubscribe or delete mod (if you used manual install) **before** running script!

What does the script do:
- checks packages, paths
- installs marisa-trie
- clones fork
- switchs to 'linux_fix' branch
- applies 'fix' (hack actually)
- creates a symlink: real mod path -> game data directory

There will be several errors with applying hack - i will fix them later (need to add a file blacklist)

if you encounter problems while installing mod or during game - please, let me know about it
(it would be good if you send lua.logs for more details) 

P.S. maybe you will need to rebind 'w' and 'a' keys. By default they are used for `great works window` and `attack`

**Comment by nikku** [comment]
Could you summarize, in a nut shell, what the actual fix is about? Maybe there is certain changes that can be applied and merged in this project to make it run univerally on Windows _and_ Linux.

**Comment by nikku** [comment]
The actual diff between your fix and the `expansion2` may also be helpful: https://github.com/Azurency/CQUI_Community-Edition/compare/expansion2...azorej:linux_fix.

**Comment by nikku** [comment]
Just tested the modification on my machine. Works absolutely fine.

**Comment by azorej** [comment]
> The actual diff between your fix and the `expansion2` may also be helpful: [expansion2...azorej:linux_fix](https://github.com/Azurency/CQUI_Community-Edition/compare/expansion2...azorej:linux_fix).

Current diff isn't very accurate: there are some needless changes. I'll fix it this weekend.
What actually i did: i made a script that makes several symlinks for every mod resource.

I don't think there is a way to make the **same steam** mod run on Windows and Linux.
As far as i know, there is no way to run custom install script for mod installation on steam for Linux.
And imho, it's not a good idea to keep linux symlinks on Windows version.
 
What can we do: we can have two steam mods. One for Windows and one for Linux. 
For linux version one must run my script before deploing mod to steam.

I'll try to make a pull request to main repository this weekend.

**Comment by Azurency** [comment]
The "fix" for linux is to **lowercase all the filenames** I think (see https://github.com/CQUI-Org/cqui/issues/301)

**Comment by azorej** [comment]
> The "fix" for linux is to **lowercase all the filenames** I think (see [CQUI-Org/cqui#301](https://github.com/CQUI-Org/cqui/issues/301))

I tried to do it and it didn't work for me:
[modified file tree](https://github.com/Azurency/CQUI_Community-Edition/files/3242543/mod.txt)
[Lua.log](https://github.com/Azurency/CQUI_Community-Edition/files/3242541/lua.log.txt)

There are a lot of errors like "Error opening/reading where=, file=C:\\Emu\\AppDataParent\\Sid Meier's Civilization VI\\Mods\\CQUI_Community-Edition\\Assets\\UI\\{filename}"
